### PR TITLE
Feature/issue 6911

### PR DESCRIPTION
**Pull request by budnix**
### Context

It seems that the Walkontable Table class has a logical bug in the `isRowBeforeRenderedRows`, `isRowAfterRenderedRows`, `isColumnBeforeRenderedColumns` and `isColumnAfterRenderedColumns` method in conjunction with hidden indexes. The methods among other things, are responsible for logic behind `getCell` method. 

I adjusted those methods to support header coordinates and updated documentation for them. Generally, the fix is about adding an additional condition for checking negative values (header coordinates). The presence of the header existence can be checked only in a case when the overlay is rendered from an index 0 (first renderable index) or if there are no rows or columns (first renderable index is then equal to -1). When the overlay is enabled by `fixedRowsTop` and counterparts options then the master overlay is shifted inwards so its first renderable index starts from 2 (depends on `fixedRowsTop` etc. settings). Hence, the master overlay doesn't have headers - in reality, they are hidden and offset from the top-left position (side effect).

### How has this been tested?

Added E2E tests.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. #6911
2.
3.

### Checklist:

- [x] My code follows the code style of this project,
- [ ] My change requires a change to the documentation.

### CODE REVIEW

File: src/3rdparty/walkontable/src/table.js [hunk 1]
```diff
@@ -730,129 +730,199 @@ class Table {
...
 
-    return level > (columnHeadersCount - 1);
+    return Math.abs(row) <= columnHeadersCount;
   }
 
+  /* eslint-disable jsdoc/require-description-complete-sentence */
   /**
-   * Check if the given row index is smaller than the index of the first row that is currently redered
-   * and return TRUE in that case, or FALSE otherwise.
+   * Check if the given row index is smaller than the index of the first row that
```
Comment (wszymanski) [inline src/3rdparty/walkontable/src/table.js:768]:
```suggestion
   * Check if the given row index is lower than the index of the first row that
```

File: src/3rdparty/walkontable/src/table.js [hunk 2]
```diff
@@ -730,129 +730,199 @@ class Table {
...
 
-  isRowAfterViewport(row) {
-    return this.rowFilter && (row > this.getLastVisibleRow());
-  }
-
+  /* eslint-disable jsdoc/require-description-complete-sentence */
   /**
-   * Check if the given column index is larger than the index of the last column that is currently redered
-   * and return TRUE in that case, or FALSE otherwise.
+   * Check if the given column index is larger than the index of the last column that
```
Comment (wszymanski) [inline src/3rdparty/walkontable/src/table.js:808]:
```suggestion
   * Check if the given column index is greater than the index of the last column that
```

File: src/3rdparty/walkontable/src/table.js [hunk 3]
```diff
@@ -730,129 +730,199 @@ class Table {
...
 
-  isColumnBeforeViewport(column) {
-    return this.columnFilter && (this.columnFilter.sourceToRendered(column) < 0 && column >= 0);
-  }
-
+  /* eslint-disable jsdoc/require-description-complete-sentence */
   /**
-   * Check if the given column index is smaller than the index of the first column that is currently redered
-   * and return TRUE in that case, or FALSE otherwise.
+   * Check if the given column index is smaller than the index of the first column that
```
Comment (wszymanski) [inline src/3rdparty/walkontable/src/table.js:843]:
```suggestion
   * Check if the given column index is lower than the index of the first column that
```

File: src/3rdparty/walkontable/src/table.js [hunk 4]
```diff
@@ -730,129 +730,199 @@ class Table {
...
    * @returns {boolean}
    */
+  /* eslint-enable jsdoc/require-description-complete-sentence */
   isColumnBeforeRenderedColumns(column) {
     const first = this.getFirstRenderedColumn();
 
-    if (column < 0) {
-      column = 0;
+    // Check the headers only in case when the first rendered column is -1 or 0.
+    // This is an indication that the overlay is placed on the most top-left position.
```
Comment (wszymanski) [inline src/3rdparty/walkontable/src/table.js:872]: `is placed on the most top-left position.`

Not just on the left one?

File: src/3rdparty/walkontable/src/table.js [hunk 5]
```diff
@@ -730,129 +730,199 @@ class Table {
...
 
-  isColumnAfterViewport(column) {
-    return this.columnFilter && (column > this.getLastVisibleColumn());
-  }
-
+  /* eslint-disable jsdoc/require-description-complete-sentence */
   /**
-   * Check if the given column index is larger than the index of the last column that is currently redered
-   * and return TRUE in that case, or FALSE otherwise.
+   * Check if the given column index is larger than the index of the last column that
```
Comment (wszymanski) [inline src/3rdparty/walkontable/src/table.js:882]:
```suggestion
   * Check if the given column index is greater than the index of the last column that
```

File: src/3rdparty/walkontable/src/table.js [hunk 6]
```diff
@@ -730,129 +730,199 @@ class Table {
...
+   *        0  ┇    ┇    ┇    ┇ <--- For fixedRowsTop: 1            │
+   *           +--------------+      the master overlay do       ---+ first rendered row (index 1)
+   *        1  │ A2 │ B2 │ C2 │      not render the first rows      │
+   *           +--------------+                                     │ FALSE
+   *        2  │ A3 │ B3 │ C3 │                                     │
+   *           +--------------+                                  ---+ last rendered row
+   *                                                                │
+   *                                                                │ TRUE
    *
    * @param {nunber} row The visual row index.
```
Comment (wszymanski) [inline src/3rdparty/walkontable/src/table.js:833]: Could you change it to the `number` by the way?

File: src/3rdparty/walkontable/src/table.js (hunk 4 above)
Comment (budnix) [reply src/3rdparty/walkontable/src/table.js]: Fixed within https://github.com/handsontable/handsontable/pull/7006/commits/4d711c8bd24bcc82016fb6327d947e60851aed64.

**Pull request merged**